op-chain-ops/cmd/celo-migrate/main.go

```diff
@@ -0,0 +1,456 @@
+			}
+			config.SetDeployments(deployments)
+
+			// Get latest block information from L1
+			l1RPC := ctx.String("l1-rpc")
+			if l1RPC == "" {
+				return fmt.Errorf("must specify --l1-rpc")
+			}
+
+			outfileL2 := ctx.Path("outfile.l2")
+			if outfileL2 == "" {
+				return fmt.Errorf("must specify --outfile.l2")
+			}
+
+			outfileRollup := ctx.Path("outfile.rollup")
+			if outfileRollup == "" {
+				return fmt.Errorf("must specify --outfile.rollup")
+			}
+
+			var l1StartBlock *types.Block
+			client, err := ethclient.Dial(l1RPC)
+			if err != nil {
+				return fmt.Errorf("cannot dial %s: %w", l1RPC, err)
+			}
+
+			if config.L1StartingBlockTag == nil {
+				l1StartBlock, err = client.BlockByNumber(context.Background(), nil)
+				if err != nil {
+					return fmt.Errorf("cannot fetch latest block: %w", err)
+				}
+				tag := rpc.BlockNumberOrHashWithHash(l1StartBlock.Hash(), true)
+				config.L1StartingBlockTag = (*genesis.MarshalableRPCBlockNumberOrHash)(&tag)
+			} else if config.L1StartingBlockTag.BlockHash != nil {
+				l1StartBlock, err = client.BlockByHash(context.Background(), *config.L1StartingBlockTag.BlockHash)
+				if err != nil {
+					return fmt.Errorf("cannot fetch block by hash: %w", err)
+				}
+			} else if config.L1StartingBlockTag.BlockNumber != nil {
+				l1StartBlock, err = client.BlockByNumber(context.Background(), big.NewInt(config.L1StartingBlockTag.BlockNumber.Int64()))
+				if err != nil {
+					return fmt.Errorf("cannot fetch block by number: %w", err)
+				}
+			}
+
+			// Ensure that there is a starting L1 block
+			if l1StartBlock == nil {
+				return fmt.Errorf("no starting L1 block")
+			}
+
+			// Sanity check the config. Do this after filling in the L1StartingBlockTag
+			// if it is not defined.
+			if err := config.Check(); err != nil {
+				return err
+			}
+
+			log.Info("Using L1 Start Block", "number", l1StartBlock.Number(), "hash", l1StartBlock.Hash().Hex())
+
+			// Build the L2 genesis block
+			l2Genesis, err := genesis.BuildL2Genesis(config, l1StartBlock)
+			if err != nil {
+				return fmt.Errorf("error creating l2 genesis: %w", err)
+			}
+
+			// Write changes to state to actual state database
+			dbPath := ctx.String("db-path")
+			if dbPath == "" {
+				return fmt.Errorf("must specify --db-path")
+			}
+			dbCache := ctx.Int("db-cache")
+			dbHandles := ctx.Int("db-handles")
+			dryRun := ctx.Bool("dry-run")
+			// TODO(pl): Move this into the function
+			log.Info("Opening database", "dbCache", dbCache, "dbHandles", dbHandles, "dbPath", dbPath)
+			ldb, err := openCeloDb(dbPath, dbCache, dbHandles)
+			if err != nil {
+				return fmt.Errorf("cannot open DB: %w", err)
+			}
+			log.Info("Loaded Celo L1 DB", "db", ldb)
+
+			cel2Header, err := ApplyMigrationChangesToDB(ldb, l2Genesis, !dryRun)
+			if err != nil {
+				return err
+			}
+
+			// Close the database handle
+			if err := ldb.Close(); err != nil {
+				return err
+			}
```

Comment:
Probably safer to defer this, or you could move the open and close into `ApplyMigrationChangesToDB`